Mathlib/GroupTheory/Solvable.lean

```diff
@@ -0,0 +1,241 @@
+/-
+Copyright (c) 2021 Jordan Brown, Thomas Browning, Patrick Lutz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jordan Brown, Thomas Browning, Patrick Lutz
+
+! This file was ported from Lean 3 source module group_theory.solvable
+! leanprover-community/mathlib commit dc6c365e751e34d100e80fe6e314c3c3e0fd2988
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Fin.VecNotation
+import Mathlib.GroupTheory.Abelianization
+import Mathlib.GroupTheory.Perm.ViaEmbedding
+import Mathlib.GroupTheory.Subgroup.Simple
+import Mathlib.SetTheory.Cardinal.Basic
+
+/-!
+# Solvable Groups
+
+In this file we introduce the notion of a solvable group. We define a solvable group as one whose
+derived series is eventually trivial. This requires defining the commutator of two subgroups and
+the derived series of a group.
+
+## Main definitions
+
+* `derivedSeries G n` : the `n`th term in the derived series of `G`, defined by iterating
+    `general_commutator` starting with the top subgroup
+* `IsSolvable G` : the group `G` is solvable
+-/
+
+
+open Subgroup
+
+variable {G G' : Type _} [Group G] [Group G'] {f : G →* G'}
+
+section derivedSeries
+
+variable (G)
+
+/-- The derived series of the group `G`, obtained by starting from the subgroup `⊤` and repeatedly
+  taking the commutator of the previous subgroup with itself for `n` times. -/
+def derivedSeries : ℕ → Subgroup G
+  | 0 => ⊤
+  | n + 1 => ⁅derivedSeries n, derivedSeries n⁆
+#align derived_series derivedSeries
+
+@[simp]
+theorem derivedSeries_zero : derivedSeries G 0 = ⊤ :=
+  rfl
+#align derived_series_zero derivedSeries_zero
+
+@[simp]
+theorem derivedSeries_succ (n : ℕ) :
+    derivedSeries G (n + 1) = ⁅derivedSeries G n, derivedSeries G n⁆ :=
+  rfl
+#align derived_series_succ derivedSeries_succ
+
+-- porting note: had to provide inductive hypothesis explicitly
+theorem derivedSeries_normal (n : ℕ) : (derivedSeries G n).Normal := by
+  induction' n with n ih
+  · exact (⊤ : Subgroup G).normal_of_characteristic
+  · exact @Subgroup.commutator_normal G _ (derivedSeries G n) (derivedSeries G n) ih ih
+#align derived_series_normal derivedSeries_normal
+
+-- porting note: removed simp attribute since `derivedSeries G 1` is not in simp-normal form
```

Comment:
Does `simp` simplify LHS to RHS? If not, then you may want to add `@[simp 1100]` to restore Lean 3 behavior.